flakes/k8s-controller-tools/flake.nix

```diff
@@ -0,0 +1,56 @@
+{
+    inputs = {
+        nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+        flake-utils.url = "github:numtide/flake-utils";
+    };
+
+    outputs = { self, nixpkgs, flake-utils }:
+        flake-utils.lib.eachSystem [ "x86_64-linux" "aarch64-darwin" ] (system:
+            let
+                pkgs = nixpkgs.legacyPackages.${system};
+
+                pname = "k8s-controller-tools";
+                version = "0.16.1";
+
+                src = pkgs.fetchFromGitHub {
+                    owner = "kubernetes-sigs";
+                    repo = "controller-tools";
+                    rev = "v${version}";
+                    sha256 = "0czfaaaxs7v7a7hy3cy83ph10r2lk4hlfn75whzp2njms5krvxh4";
+                };
+
+                k8s-controller-tools = pkgs.buildGoModule rec {
```

Comment:
`buildGoModule` ifs the nix function that does most of the magic of building the actual go program:
https://nixos.org/manual/nixpkgs/stable/#ssec-language-go

About patches:
https://nixos.org/manual/nixpkgs/stable/#var-stdenv-patches